### DISCUSSION TITLE
How to handle different system architecture [arm/x86]?

### DISCUSSION THREAD

**Discussion by FridayPush** [DavidVujic/python-polylith#187] [Q&A] [answered]
Does the current workflow support the building of packages for architectures that differ from the host environment?

Scenario: Building a project that uses a Docker container and will run in an x86 environment, from an M1 Mac. 

Some current examples: 
Docker offers flags related to platform:
`docker build --platform linux/amd64`
Pip directly offers the same:
`pip install -r requirements.txt --platform linux_x86_64`

**Comment by DavidVujic** [comment, marked as answer]
Hi @FridayPush! Polylith runs on top of tools like Poetry, Hatch, PDM and Rye. The packaging capabilities is depending on which one of those tools you are using. Polylith doesn't deal with the actual building of a wheel, but will collect code before a build action is triggered. For Poetry, it is done with a plugin and for Hatch, PDM and Rye that is accomplished with build hooks as described in the docs.

I hope this answer is helpful!
